test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,285 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <memory>
+#include <random>
+#include <utility>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/search/algorithm/all.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <stdlib.h>
+
+using namespace seqan3;
+using namespace seqan3::test;
+
+struct Options
+{
+    size_t const sequence_length;
+    bool const repeating;
+    size_t const number_of_reads;
+    size_t const read_length;
+    double const prob_insertion;
+    double const prob_deletion;
+    double const simulated_errors;
```

Comment:
```suggestion
    uint8_t const simulated_errors;
```
?

Reply:
simulated_errors is value that can be given to normal distribution in the function generate_reads. (I changed the Benchmark parameters in the last commit the actually get a double now.)

Reply:
Not resolved

Reply:
Still open.

Either this is an integer for the number of errors, or a *argument* to `std::normal_distribution`. So I don't see any reason this has to be double.

Reply:
Hm the second argument for normal distribution is the Standard deviation which does not have to be an integer. I could rewrite the function to get either get a interger number of errors to simulate or the double for the standard deviation, which maybe would make it clearer.

Reply:
So during the implementation I noticed that the expected type of normal distribution is double so I have to cast it on the input parameter simulated_errors errors now. But overall it should be clearer now what actually happens.